aws/resource_aws_elasticache_parameter_group.go

```diff
@@ -188,6 +188,100 @@ func resourceAwsElasticacheParameterGroupUpdate(d *schema.ResourceData, meta int
 				}
 				return nil
 			})
+
+			// When attempting to reset the reserved-memory parameter, the API
+			// can return the below 500 error, which causes the AWS Go SDK to
+			// automatically retry and hence timeout resource.Retry():
+			//   InternalFailure: An internal error has occurred. Please try your query again at a later time.
+			// Instead of hardcoding the reserved-memory parameter removal
+			// above, which may become out of date, here we add logic to
+			// workaround this API behavior
+
+			if isResourceTimeoutError(err) {
+				for i, paramToModify := range paramsToModify {
+					if aws.StringValue(paramToModify.ParameterName) != "reserved-memory" {
+						continue
+					}
+
+					// Always reset the top level error and remove the reset for reserved-memory
+					err = nil
+					paramsToModify = append(paramsToModify[:i], paramsToModify[i+1:]...)
+
+					// If we are only trying to remove reserved-memory and not perform
+					// an update to reserved-memory or reserved-memory-percentage, we
+					// can attempt to workaround the API issue by switching it to
+					// reserved-memory-percentage first then reset that temporary parameter.
+
+					// The reserved-memory-percentage parameter does not exist in redis2.6 and redis2.8
+					family := d.Get("family").(string)
+					if family == "redis2.6" || family == "redis2.8" {
```

Comment:
We should add a log here or I'm even tempted to error out if this value has changed while using this family
```suggestion
log.Printf("[WARN] unable to update `reserved-memory-percentage` while family is %q: %s", family, err)

```

Reply:
I'll add a warning log in there 👍